### PR TITLE
ICU-22220 CLDR release-43-alpha1 to ICU main

### PR DESCRIPTION
**Pull request by pedberg-icu**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22220
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

This integrates CLDR release-43-alpha1 to  to ICU; it has several significant changes compared to alpha0.
1. New CLDR structure primarily intended for ICU4X, most of which should be excluded for ICU (at least for now)
    - For supplemental calendar data, new code and aliases attributes for eras. Also, the japanese calendar eras have two added eras with negative type values which cannot currently be supported in ICU. The regex for mapping calendar era data needed to be changed to exclude these.
    - For parentLocales, additional data for collation and segmentation parents; this is distinguished by a component attribute on the parentLocales element. It had to be excluded from the normal parentLocales map used for determining locale parents.
2. A large number of  likelySubtags additions contributed by SIL.
    - This exposed a problem in ICU4C uscript_getCode and ICU4J UScript.getCode() in which their nameOrAbbrOrLocale parameter - which can take a script code, script alias, or language tag - was interpreting script aliases such as "Old-Hungarian" or "New_Tai_Lue" as language tags, since "old" and "new" are valid language codes now mapped  n likelySubtags. The code has been updated to add more explicit checks that favor interpretation as script aliases for such cases.
3. To resolve some issues with lateral inheritance, some CLDR data values intended to be inherited have been explicitly populated with their resolved values. Although this results in more strings in the CLDR data, the ICU strings pool should reduce the memory impact on ICU.
4. Five new "Geminate" variants of Ethiopic transforms have been added. There seems to be an issue with these that prevents instantiation in ICU; that is being investigated with [CLDR-16408](https://unicode-org.atlassian.net/browse/CLDR-16408), and corresponding tests are skipped in ICU with a logKnownIssue against that CLDR ticket.

[CLDR-16408]: https://unicode-org.atlassian.net/browse/CLDR-16408?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/SupplementalData.java [hunk 1]
```diff
@@ -123,7 +124,8 @@ public static SupplementalData create(CldrDataSupplier src) {
                             alias,
                             alias.typeKey.valueFrom(v),
                             alias.replacementKey.valueFrom(v)));
-                } else if (PARENT_LOCALE.matches(v.getPath())) {
+                } else if (PARENT_LOCALE.matches(v.getPath()) && !COMPONENT.optionalValueFrom(v).isPresent()) {
+                    // CLDR-16253 added component-specific parents, which we ignore for now.
```
Comment (sffc) [inline tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/SupplementalData.java:128]: We should handle these properly. Please make this a TODO(ICU-22289) and I'll try to take that issue for icu 73

**Review by sffc — COMMENTED** [review]
Is it expected to have all the .res files checked in under the maven directory? @echeran

**Comment by srl295** [comment]
> Is it expected to have all the .res files checked in under the maven directory? @echeran

https://unicode-org.atlassian.net/browse/ICU-12968 to remove binary data from the repo

**Review by srl295 — APPROVED** [review]

File: icu4c/source/common/uscript.cpp [hunk 2]
```diff
@@ -114,7 +114,20 @@ uscript_getCode(const char* nameOrAbbrOrLocale,
...
+    // Favor interpretation of nameOrAbbrOrLocale as a script alias if either
+    // 1. nameOrAbbrOrLocale does not contain -/_. Handles Han, Mro, Nko, etc.
+    // 2. The last instance of -/_ is at offset 3, and the portion after that is
+    //    longer than 4 characters (i.e. not a script or region code). This handles
+    //    Old_Hungarian, Old_Italic, etc. ("old" is a valid language code)
+    // 3. The last instance of -/_ is at offset 7, and the portion after that is
+    //    3 characters. This handles New_Tai_Lue ("new" is a valid language code).
+    if (lastSepPtr==nullptr
+            || (lastSepPtr-nameOrAbbrOrLocale == 3 && uprv_strlen(nameOrAbbrOrLocale) > 8)
+            || (lastSepPtr-nameOrAbbrOrLocale == 7 && uprv_strlen(nameOrAbbrOrLocale) == 11) ) {
```
[the remark this answers is not in the document]
Comment (pedberg-icu) [reply icu4c/source/common/uscript.cpp:130]: @srl295 I did add a test for the various cases here, see changes in icu4c/source/test/cintltst/cucdapi.c

**Pull request merged**